client/src/partials/reports/global_transaction/global_transaction.js

```diff
@@ -133,45 +142,64 @@ angular.module('bhima.controllers')
     }
 
     function search () {
-      if (!$scope.model.account_id) { return; }
-      $scope.mode = $scope.model.account_id !== 0 ? 'selected' : 'all';
+      $scope.mode = ($scope.model.account_id && $scope.model.account_id > 0) ? 'selected' : 'all';
+      
+      if ($scope.model.account_id && $scope.model.account_id > 0) {
+        $scope.model.account_number = $scope.accounts.get($scope.model.account_id).account_number;       
+      } else {
+        $scope.model.account_number = 0;
+      }
+
       var qo = {
         source : $scope.model.source_id,
         enterprise_id : $scope.enterprise.id,
         account_id : $scope.model.account_id,
-        datef : util.sqlDate($scope.state.from),
-        datet : util.sqlDate($scope.state.to)
+        datef : util.sqlDate($scope.dates.from),
+        datet : util.sqlDate($scope.dates.to)
       };
 
-      if ($scope.model.account_id && $scope.model.account_id === 0) {
-        $scope.model.account_number = 'Tous';
-      } else {
-        $scope.model.account_number = $scope.accounts.get($scope.model.account_id).account_number;
-      }
-
       connect.fetch('/reports/allTrans/?'+JSON.stringify(qo))
       .then(function (res) {
         if (res.length > 0) {
```

Comment:
You do not need to check if res.length > 0.  If res is an empty array, `res.map()` will not be called.